src/extras/ShapeUtils.js

```diff
@@ -21,670 +21,878 @@ var ShapeUtils = {
 
 	},
 
-	triangulate: ( function () {
+	isClockWise: function ( pts ) {
 
-		/**
-		 * This code is a quick port of code written in C++ which was submitted to
-		 * flipcode.com by John W. Ratcliff  // July 22, 2000
-		 * See original code and more information here:
-		 * http://www.flipcode.com/archives/Efficient_Polygon_Triangulation.shtml
-		 *
-		 * ported to actionscript by Zevan Rosser
-		 * www.actionsnippet.com
-		 *
-		 * ported to javascript by Joshua Koo
-		 * http://www.lab4games.net/zz85/blog
-		 *
-		 */
+		return ShapeUtils.area( pts ) < 0;
 
-		function snip( contour, u, v, w, n, verts ) {
+	},
 
-			var p;
-			var ax, ay, bx, by;
-			var cx, cy, px, py;
+	triangulate: function ( vertices, holeIndices ) {
 
-			ax = contour[ verts[ u ] ].x;
-			ay = contour[ verts[ u ] ].y;
+		// Port from: https://github.com/mapbox/earcut (v2.1.2)
 
-			bx = contour[ verts[ v ] ].x;
-			by = contour[ verts[ v ] ].y;
+		function earcut( data, holeIndices, dim ) {
```

Comment:
Probably better to move this function to the top of the file? Otherwise we're going to be instantiating it every time we call to `triangulate`.

Reply:
Okay, but i think we have to move all functions (not only `earcut`) inside `triangulate` because they are on the same hierarchy. How about moving to code to a new file `Earcut.js` in `src/extras` instead? In this way, the code is a little bit better separated and organized.

Reply:
I've made a PR to see the result of this suggestions.